MotionPhotoReader/app/src/main/java/com/example/motionphotoreader/MotionPhotoInfo.java

```diff
@@ -0,0 +1,109 @@
+package com.example.motionphotoreader;
+
+import android.media.MediaExtractor;
+import android.media.MediaFormat;
+
+import com.adobe.internal.xmp.XMPException;
+import com.adobe.internal.xmp.XMPMeta;
+
+import java.io.File;
+import java.io.FileDescriptor;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+/**
+ * Holds data about the Motion Photo file.
+ */
+public class MotionPhotoInfo {
+
+    private final int width;
+    private final int height;
+    private final long duration;
+
+    private final int videoOffset;
+    private final long presentationTimestampUs;
+
+    private MediaExtractor extractor;
+
+    /**
+     * Creates a MotionPhotoInfo object associated with a given file.
+     */
+    private MotionPhotoInfo(MediaFormat mediaFormat, int videoOffset, long presentationTimestampUs) {
```

Comment:
add `MediaExtractor extractor` to constructor argument, and then set `this.extractor = extractor`

Reply:
I don't actually reference the extractor anywhere else in the class where it isn't provided the extractor via an argument. Would it be okay to just delete the extractor field?